### PR TITLE
Airflow support for GCP 

### PR DESCRIPTION
**Pull request by valayDave**
## Task List : 

- [x] Test the integration using valayDave/metaflow#67
- [x] Document the introduction of the `deploy_airflow` and `deploy_argo` flags
- [x] Document the types of resources deployed via the PR 
- [x] Add the deployment-related information to the Airflow Eng Infra Docs. 
- [x] Remove Log shipping to GCS before Merge. Not needed ATM since we are running a local scheduler.

### CODE REVIEW

File: gcp/terraform/services/airflow.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  count = var.deploy_airflow ? 1 : 0
+  api_version = "v1"
+  kind        = "ServiceAccount"
+  metadata {
+    name      = "airflow-deployment-scheduler"
+    namespace = kubernetes_namespace.airflow[0].metadata[0].name
+  }
+  annotations = {
+    "iam.gke.io/gcp-service-account" = "${var.metaflow_workload_identity_gsa_name}@${var.project}.iam.gserviceaccount.com"
+  }
```
Comment (valayDave) [inline gcp/terraform/services/airflow.tf:45]: TODO : Add `dependsOn` over here to depend on the helm deployment for creating the annotation.

File: gcp/terraform/README.md [hunk 2]
```diff
@@ -50,6 +50,35 @@ Then, apply the `services` module (deploys Metaflow services to GKE)
...
 
+## Metaflow job orchestration options
+The recommended way to orchestrate Metaflow workloads on Kubernetes is via [Argo Workflows](https://docs.metaflow.org/going-to-production-with-metaflow/scheduling-metaflow-flows/scheduling-with-argo-workflows). However, Airflow is also supported as an alternative.
+
+The template also provides the `deploy_airflow` and `deploy_argo` flags as variables. These are booleans that specify if [Airflow](https://airflow.apache.org/) or [Argo Workflows](https://argoproj.github.io/argo-workflows/) will be deployed in the Kubernetes cluster along with Metaflow related services. By default `deploy_argo` is set to __true__ and `deploy_airflow` is set to __false__.
+To change these, set them in your `FILE.tfvars` file (or else, via other [terraform variable](https://www.terraform.io/language/values/variables) passing mechanisms)
+
+### Argo Workflows
+Argo Workflows is installed by default on the AKS cluster as part of the `services` submodule. Setting the `deploy_argo` [variable](./variables.tf) will deploy Argo in the AKS cluster. Not additional configuration is done in the `infra` module to support `argo`.
```
Comment (jackie-ob) [inline gcp/terraform/README.md:60]: AKS / GKE

Comment (jackie-ob) [reply gcp/terraform/README.md:60]: "Not additional configuration" / "No additional configuration"

File: gcp/terraform/README.md [hunk 3]
```diff
@@ -50,6 +50,35 @@ Then, apply the `services` module (deploys Metaflow services to GKE)
...
+**This is quickstart template only, not recommended for real production deployments**
+
+If `deploy_airflow`  is set to true, then the `services` module will deploy Airflow via a [helm chart](https://airflow.apache.org/docs/helm-chart/stable/index.html) into the kubernetes cluster (the one deployed by the `infra` module). 
+
+The terraform template deploys Airflow configured with a `LocalExecutor`. Metaflow can work with any Airflow executor. This template deploys the `LocalExecutor` for simplicity.
+
+After you have changed the value of `deploy_airflow`, reapply terraform for both [infra and services](#usage).
+
+#### Shipping Metaflow compiled DAGs to Airflow
+Airflow expects Python files with Airflow dags present in the [dags_folder](https://airflow.apache.org/docs/apache-airflow/2.2.0/configurations-ref.html#dags-folder). By default this terraform template uses the [defaults](https://airflow.apache.org/docs/helm-chart/stable/parameters-ref.html#airflow) set in the Airflow helm chart which is `{AIRFLOW_HOME}/dags` (`/opt/airflow/dags`).
```
Comment (jackie-ob) [inline gcp/terraform/README.md:75]: "DAGs" all caps?

File: gcp/terraform/README.md [hunk 4]
```diff
@@ -50,6 +50,35 @@ Then, apply the `services` module (deploys Metaflow services to GKE)
...
+
+The metaflow-tools repository also ships a [airflow_dag_upload.py](../../scripts/airflow_dag_upload.py) file that can help sync Airflow dag files generated by Metaflow to the Airflow scheduler _deployed by this template_. Under the hood [airflow_dag_upload.py](../../scripts/airflow_dag_upload.py) uses the `kubectl cp` command to copy files from local to the Airflow scheduler's container. Example of how to use the file:
```
Comment (jackie-ob) [inline gcp/terraform/README.md:77]: file/script

File: gcp/terraform/output.tf [hunk 5]
```diff
@@ -76,7 +78,7 @@ $ kubectl port-forward -n argo deployment/argo-server 2746:2746
 
 option 2 - this script manages the same port-forwards for you (and prevents timeouts)
 
-$ python metaflow-tools/scripts/forward_metaflow_ports.py [--include-argo]
+$ python metaflow-tools/scripts/forward_metaflow_ports.py [--include-argo][--include-airflow]
```
Comment (jackie-ob) [inline gcp/terraform/output.tf:81]: missing a space between the two options.

**Review by jackie-ob — APPROVED** [review]

**Pull request merged**